wrapper/core/mcserver.py

```diff
@@ -762,12 +762,20 @@ def readconsole(self, buff):
             major = get_int(getargs(semantics, 1))
             minor = get_int(getargs(semantics, 2))
             self.vitals.version_compute = minor + (major * 100) + (release * 10000)  # noqa
-
+            
+            #l33tlinuxh4x0r's snapshot fix.
+            if "." not in self.vitals.version:
+                self.log.warning("\nYou are running a Minecraft Snapshot "
+                                 "\nExpect odd behavior!!! ")
+                self.vitals.version_compute = -1
+                
             # 1.7.6 (protocol 5) is the cutoff where ops.txt became ops.json
             if self.vitals.version_compute > 10705 and self.vitals.protocolVersion < 0:  # noqa
                 self.vitals.protocolVersion = 5
-                self.wrapper.api.registerPermission("mc1.7.6", value=True)
-            if self.vitals.version_compute < 10702 and self.wrapper.proxymode:
+            # commenting out the following line fixes items on ground not being seen in proxy mode.
+            #    self.wrapper.api.registerPermission("mc1.7.6", value=True)
+            # updated the following line to account for the snapshot fix.
```

Comment:
This really does not fix the items on the ground; all it is doing it disabling proxy mode.  We need to find out why items are not showing in the first place.  The "mc1.7.6" is only used by console for running ban commands at this time.  Setting the version compute to -1 is what "fixes" it and disables proxy (not a real solution since you could just turn off proxy if that's what you wanted).

I will have to review changes to the snapshot.  Actually, MCserver.py is not the place you should be looking to bypass this.  The code for proxy that decides not to allow a player on because of a is in clientconnection.py.  The drops problem might actually be in the SPAWN_OBJECT packet in parse_sb.py

Reply:
Also, if you can put a print statement in 771 so we can see what the snaps shots "versionCompute" is..

Reply:
I tried this on version 1.12.2 and proxy is in fact running. All that the -1 does is allow all versions of snapshots with the current snapshot scheme to run proxy mode even thought it doesn't work packet wise. Just a test. Also it throws a warning that alerts the server owner that it is probably going to be bugged. I will comment out the -1 code and try with just the commented `self.wrapper.api.registerPermission("mc1.7.6", value=True)` line and report back.

Reply:
Also line `if self.vitals.version_compute < 10702 and self.vitals.version_compute > 0 and self.wrapper.proxymode:` is changed... The `and self.vitals.version_compute > 0` allows snapshots which I set version to -1 to be allowed to run proxy mode.

Reply:
I just tested with 1.12.2 and it seems to work intermittently with just `self.wrapper.api.registerPermission("mc1.7.6", value=True)` being commented and no other changes. But it works more often than when it is not commented out. Not sure but it may be due to later versions of Minecraft using a newer version than 1.7.6 for entities? Might have to put in some code to determine what version of Minecraft protocol is being used.

Reply:
I think it is something else entirely.  That permission is used _solely_ for determining what version of the ban command to run.  Nothing in the proxy connection uses this permission (and that is all it is, a permission node).
The only reason a snapshot works with the version being set to -1 is because you are disabling proxy mode.

If you were in proxy mode, you could never connect with a -1:

This is the proxy logon code in clientconnection.py (which you have not changed).  It will not let you login if the versions (client and server) are not the same or if protocolVersion is set to -1:
```code: python
    def _parse_handshaking(self):
        # "version|address|port|state"
        data = self.packet.readpkt([VARINT, STRING, USHORT, VARINT])

        self.clientversion = data[0]
        self._getclientpacketset()
                    ...
            if self.servervitals.protocolVersion == -1:  <=============HERE
                #  ... returns -1 to signal no server
                self.disconnect(
                    "The server is not started (protocol not established)."
                )
                return False

                    ...

            if self.servervitals.protocolVersion != self.clientversion: <====HERE
                self.disconnect("You're not running the same Minecraft"
                                " version as the server!")
                return False
                    ...
```
These parts of code you are looking at in mcserver.py do no control proxy mode.  They are only used for the wrapped server in the console.

You are not in proxy mode.  You can tell when a player connects.  This is what the console will read if you connect with no proxy:
```
[14:38:52] [Server thread/INFO]: SurestTexas00[/127.0.0.1:59542] logged in with entity id 251 at (138.33847601794923, 55.0, 152.1111576647758)
[14:38:53] [Server thread/INFO]: SurestTexas00 joined the game
```
<br>
This is what a proxy logon will look like (the second line will only show up if debug is set in the logging.json):

```
[14:36:29] [requests.packages.urllib3.connectionpool/INFO]: Starting new HTTPS connection (1): sessionserver.mojang.com
[14:36:32] [requests.packages.urllib3.connectionpool/DEBUG]: "GET /session/minecraft/hasJoined?username=SurestTexas00&serverId=-771c4cd8b64fe0bad90627d8f716b22d0775883a HTTP/1.1" 200 1125
[14:36:32] [Wrapper.py/INFO]: SurestTexas00's Proxy Client LOGON occurred: (UUID: 4f953255-6775-4dc6-a612-fb4230588eff | IP: 127.0.0.1 | SecureConnection: True)
[14:36:32] [Server thread/INFO]: SurestTexas00[/127.0.0.1:56374] logged in with entity id 0 at (1.4039747385308137, 4.0, 3.7415010830076687)
[14:36:32] [Server thread/INFO]: SurestTexas00 joined the game
```